core/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/SameShardAllocationDecider.java

```diff
@@ -45,15 +47,22 @@
 
     public static final String NAME = "same_shard";
 
-    public static final String SAME_HOST_SETTING = "cluster.routing.allocation.same_shard.host";
+    public static final Setting<Boolean> CLUSTER_ROUTING_ALLOCATION_SAME_HOST_SETTING =
+        Setting.boolSetting("cluster.routing.allocation.same_shard.host", false,
+            Setting.Property.Dynamic, Setting.Property.NodeScope);
```

Comment:
Right. I wondered if it should be dynamic or not.
The reason I made it dynamic is [document](https://www.elastic.co/guide/en/elasticsearch/reference/master/shards-allocation.html) says it's dynamic.
(and thought making it dynamic maybe useful)

I'll make it non-dynamic.